host-bin/mount-chroot

```diff
@@ -216,32 +228,78 @@ $passphrase" | ecryptfs-wrap-passphrase "$wrappedfnek" -
             tail -c 160 "$KEYFILE" | head -c 80 > "$wrappedkey"
             tail -c 80 "$KEYFILE" > "$wrappedfnek"
 
+            PAD1="`head -c 16 "$wrappedkey"`"
+            PAD2="`tail -c 16 "$wrappedfnek"`"
+
+            if [ "$PAD1" = '................' ] && [ "$PAD2" = '................']; then
+                NOENC='y'
+                key="`tail -c 64 "$wrappedkey"`"
+                fnek="`head -c 64 "$wrappedfnek"`"
+            else
+                echo -n "Enter encryption passphrase for $NAME: " 1>&2
+                [ -t 0 ] && stty -echo
+                if [ -z "$passphrase" ]; then
+                    read -r passphrase
+                fi
+                [ -t 0 ] && stty echo
+                echo 1>&2
+            fi
+
             # Change the passphrase if requested
             if [ "${ENCRYPT:-0}" -ge 2 ]; then
                 oldpassphrase="$passphrase"
                 passphrase="$CROUTON_NEW_PASSPHRASE"
                 promptNewPassphrase
-
                 echo "Applying passphrase change" 1>&2
-                echo -n "$oldpassphrase
+
+                if [ -z "$NOENC" -a -z "$NOPASS" ]; then
+                    echo -n "$oldpassphrase
 $passphrase" | ecryptfs-rewrap-passphrase "$wrappedkey" -
-                echo -n "$oldpassphrase
+                    echo -n "$oldpassphrase
 $passphrase" | ecryptfs-rewrap-passphrase "$wrappedfnek" -
-                echo | cat - "$wrappedkey" "$wrappedfnek" > "$KEYFILE"
+                    echo | cat - "$wrappedkey" "$wrappedfnek" > "$KEYFILE"
+                elif [ -n "$NOENC" -a -z "$NOPASS" ]; then
+                    echo -n "$key
+$passphrase" | ecryptfs-wrap-passphrase "$wrappedkey" -
+                    echo -n "$fnek
+$passphrase" | ecryptfs-wrap-passphrase "$wrappedfnek" -
+                    echo | cat - "$wrappedkey" "$wrappedfnek" > "$KEYFILE"
+                    unset key fnek
+                elif [ -z "$NOENC" -a -n "$NOPASS" ]; then
+                    key="`echo -n "$oldpassphrase" \
+                        | ecryptfs-unwrap-passphrase "$wrappedkey" - 2>/dev/null`"
+                    fnek="`echo -n "$oldpassphrase" \
+                        | ecryptfs-unwrap-passphrase "$wrappedfnek" - 2>/dev/null`"
+                    echo -n "
+................$key................$fnek" > "$KEYFILE"
+                elif [ -n "$NOENC" -a -n "$NOPASS" ]; then
+                    echo "Changing from nothing to nothing? You must be high-security..."
```

Comment:
I guess the point I was making with the alternate wording was that I'd rather not direct snark at users (each one of them is a special flower and whatever).